openapiart/generator.py

```diff
@@ -345,17 +345,24 @@ def _write_snappi_object(self, ref, choice_method_name=None):
                 self._write()
```

Comment:
should be following snippet:


```python
            # write _TYPES definition
            # TODO: this func won't detect whether $ref for a given property is
            # a list because it relies on 'type' attribute to do so
            snappi_types = self._get_snappi_types(schema_object)
            if len(snappi_types) > 0:
                self._write(1, '_TYPES = {')
                for name, value in snappi_types:
                    if len(value) == 1:
                        self._write(2, "'%s': {'%s': %s}," % (
                            name, list(value.keys())[0], list(value.values())[0]
                        ))
                        continue
                    self._write(2, "'%s': %s" % (name, '{'))
                    for n, v in value.items():
                        if isinstance(v, list):
                            self._write(3, "'%s': [" % n)
                            for i in v:
                                self._write(4, "'%s'," % i)
                            self._write(3, "],")
                            continue
                        self._write(3, "'%s': %s," % (n, v))
                    self._write(2, "},")
                self._write(1, '} # type: Dict[str, str]')
                self._write()
            else:
                # TODO: provide empty types as workaround because deserializer
                # in snappicommon.py currently expects it
                self._write(1, '_TYPES = {} # type: Dict[str, str]')
                self._write()
```